site/cypress/e2e/pdfs.spec.js

```diff
@@ -0,0 +1,102 @@
+function switch_settings(gradeable_id) {
+    cy.visit(['sample', 'gradeable', gradeable_id, 'update']);
+    cy.get('#page_3_nav').click();
+    cy.get('input[data-testname="grader_assignment_method"][value="1"]').check();
+}
+
+function revert_settings(gradeable_id, setting) {
+    cy.visit(['sample', 'gradeable', gradeable_id, 'update']);
+    cy.get('#page_3_nav').click();
+    cy.get(`input[data-testname="grader_assignment_method"][id=${setting}`).check();
+}
+
+function select_gradeable() {
+    // This gets a gradeable that has been graded already, so there are submissions available.
+    cy.get('[data-testid="grade-table"]').contains('/ 12').click({ force: true });
+    cy.get('[data-testid="show-autograding"]').click();
+    cy.get('[data-testid="show-submission"]').click();
+    cy.get('[data-testid="folders"]').contains('submissions').click();
+    cy.get('#div_viewer_sd1').contains('words_').click();
+    cy.get('#pageContainer1').should('be.visible');
+}
+
+function check_pdf_access(gradeable_id) {
+    cy.visit(['sample', 'gradeable', gradeable_id, 'grading', 'details']);
+    cy.get('#agree-button').click({ force: true });
+    cy.get('[data-testid="details-table"]').should('be.visible');
+    cy.get('[data-testid="view-sections"]').then(($button) => {
+        if ($button.text().includes('View All')) {
+            $button.click();
+        }
+    });
+    select_gradeable();
+}
+
+function minimum_pdf_access(gradeable_id) {
+    cy.visit(['sample', 'gradeable', gradeable_id, 'grading', 'details']);
+    cy.get('#agree-button').click({ force: true });
+    cy.get('[data-testid="details-table"]').should('be.visible');
+    select_gradeable();
+}
+
+function no_pdf_access(gradeable_id) {
+    cy.visit(['sample', 'gradeable', gradeable_id, 'grading', 'details']);
+    cy.get('[data-testid="banner-message"]').should('be.visible').and('contain.text', 'You do not have permission');
+}
+
+const gradeable_type = ['grading_homework_team_pdf', 'grading_homework_pdf', 'grading_pdf_peer_homework', 'grading_pdf_peer_team_homework'];
```

Comment:
Just a comment: it is sometimes cleaner to format arrays with each element on a separate row so it's easier to read a diff if someone changes one single element in the future.

Reply:
This has been added.